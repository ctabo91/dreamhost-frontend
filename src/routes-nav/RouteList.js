import React, { useContext } from "react";
import { Route, Routes, Navigate } from "react-router-dom";
import Home from "../homepage/Home";
import RecipesList from "../meals-drinks/RecipesList";
import RecipeDetails from "../meals-drinks/RecipeDetails";
import CreateOrUpdateRecipeForm from "../meals-drinks/CreateOrUpdateRecipeForm";
import LoginForm from "../auth/LoginForm";
import SignupForm from "../auth/SignupForm";
import Profile from "../profiles/Profile";
import UserContext from "../auth/UserContext";
import CategoryList from "../meals-drinks/CategoryList";


/** Site-wide routes.
 * 
 * Certain sites should be restricted, unless logged in.
 * 
 * Visiting these restricted routes when not logged in redirects to the login page.
 */ 

function RouteList({ login, signup, hasFavoritedRecipe, favoriteRecipe }) {
    const { currentUser } = useContext(UserContext);
    return (
        <div>
            <Routes>

                <Route exact path="/" element={<Home />} />

                <Route exact path="/login" element={<LoginForm login={login} />} />

                <Route exact path="/signup" element={<SignupForm signup={signup} />} />            

                {currentUser ? (
                    <>
                        <Route 
                            exact
                            path="/recipes/:item"
                            element={<RecipesList />}
                        />

                        <Route 
                            exact
                            path="/recipes/:item/categories"
                            element={<CategoryList />}
                        />

                        <Route 
                            exact 
                            path="/recipes/:item/:category"
                            element={<RecipesList />}
                        />

                        <Route 
                            exact 
                            path="/recipes/details/:access/:item/:id"
                            element={<RecipeDetails hasFavoritedRecipe={hasFavoritedRecipe} favoriteRecipe={favoriteRecipe} />}
                        />

                        <Route 
                            exact 
                            path="/recipes/create/:item" 
                            element={<CreateOrUpdateRecipeForm />} 
                        />
                        

                        <Route 
                            exact 
                            path="/recipes/update/:item/:id"
                            element={<CreateOrUpdateRecipeForm />}
                        />

                        <Route 
                            exact 
                            path="/profile"
                            element={<Profile />}
                        />
                    </>
                ) : (
                    <>
                        <Route 
                            exact
                            path="/recipes/:item"
                            element={<Navigate to="/login" replace />}
                        />

                        <Route 
                            exact
                            path="/recipes/:item/categories"
                            element={<Navigate to="/login" replace />}
                        />

                        <Route
                            exact
                            path="/recipes/:item/:category"
                            element={<Navigate to="/login" replace />} 
                        />

                        <Route
                            exact
                            path="/recipes/:item/:id"
                            element={<Navigate to="/login" replace />} 
                        />

                        <Route
                            exact
                            path="/recipes/create/:item"
                            element={<Navigate to="/login" replace />} 
                        />

                        <Route
                            exact
                            path="/recipes/update/:item/:id"
                            element={<Navigate to="/login" replace />} 
                        />

                        <Route
                            exact
                            path="/profile"
                            element={<Navigate to="/login" replace />} 
                        />
                    </>
                )}

            </Routes>
        </div>
    )
}

export default RouteList;