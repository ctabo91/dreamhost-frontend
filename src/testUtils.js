import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser, hasFavoritedRecipe = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasFavoritedRecipe }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };