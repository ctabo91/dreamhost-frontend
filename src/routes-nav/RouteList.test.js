import React from "react";
import { render } from "@testing-library/react";
import RouteList from "./RouteList";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <RouteList />
            </UserProvider>
        </MemoryRouter>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <RouteList />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
