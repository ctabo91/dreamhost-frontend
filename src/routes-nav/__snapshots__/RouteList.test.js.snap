// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Home mt-5 pt-5"
    >
      <h1
        class="Home-title display-1 text-center pt-5"
      >
        Welcome testfirst!
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
